extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/StaticTenantResolver.java

```diff
@@ -93,13 +96,16 @@ private DefaultStaticTenantResolver(TenantConfigBean tenantConfigBean) {
 
         @Override
         public String resolve(RoutingContext context) {
-            String[] pathSegments = context.request().path().split("/");
-            if (pathSegments.length > 0) {
-                String lastPathSegment = pathSegments[pathSegments.length - 1];
-                if (tenantConfigBean.getStaticTenant(lastPathSegment) != null) {
+            String path = context.request().path();
+            if (isNullOrEmpty(path)) {
+                return null;
+            }
+            String[] pathSegments = path.split(PATH_SEPARATOR);
+            for (String segment : pathSegments) {
+                if (isNotNullOrEmpty(segment) && tenantConfigBean.getStaticTenant(segment) != null) {
```

Comment:
Please remove this `isNotNullOrEmpty` check here as well, and the corresponding update in StringUtils